client/apollo/react/src/Button/ButtonCommon.tsx

```diff
@@ -0,0 +1,48 @@
+import type {
+  ComponentPropsWithoutRef,
+  PropsWithChildren,
+  ReactNode,
+} from "react";
+
+export const buttonVariants = {
+  primary: "primary",
+  "primary-business": "primary-business",
+  "primary-inverse": "primary-inverse",
+  secondary: "secondary",
+  "secondary-inverse": "secondary-inverse",
+  tertiary: "tertiary",
+  ghost: "ghost",
+} as const;
+
+export type ButtonVariants = keyof typeof buttonVariants;
```

Comment:
pourquoi passer par un objet ?
```suggestion
export const buttonVariants = ["primary", "primary-business", "primary-inverse", "secondary", "secondary-inverse","tertiary","ghost" 
] as const;

export type ButtonVariants = typeof buttonVariants;
```

Reply:
niveau typage ce n'est pas la même chose

Reply:
l'avantage de l'objet c'est que le consommateur peut l'utiliser un peu comme un enum mais sans les contraintes d'un enum